Sources/Sentry/SentryClient.m

```diff
@@ -638,6 +642,23 @@ - (BOOL)isOOM:(SentryEvent *)event isCrashEvent:(BOOL)isCrashEvent
         [exception.mechanism.type isEqualToString:SentryOutOfMemoryMechanismType];
 }
 
+- (void)storeFreeMemoryToDeviceContext:(SentryEvent *)event
```

Comment:
I was actually wondering if we shouldn't remove it altogether. In the case of an OOM, we'll still get the up-to-date value of free memory with this fix in place, so I think we can remove the if/else block and always call `storeFreeMemoryToDeviceContext`? What do you think?

Reply:
If we do want to keep both functions and keep the if/else in place, then I do think having both storeFreeMemoryToDeviceContext and removeFreeMemoryFromDeviceContext like this is a lot better than trying to avoid a few duplicated lines by having one function do 2 different things based on a boolean. Curious what you think.

Reply:
Shouldn't this do the trick?
```
- (void)storeFreeMemoryToDeviceContext:(SentryEvent *)event
{
    [self modifyDeviceContext:event
                        block:^(NSMutableDictionary *device) {
                            NSDictionary *systemInfo = SentryCrash.sharedInstance.systemInfo;
                            device[SentryDeviceContextFreeMemoryKey] = systemInfo[@"freeMemory"];
                        }];
}

- (void)removeFreeMemoryFromDeviceContext:(SentryEvent *)event
{
    [self modifyDeviceContext:event
                        block:^(NSMutableDictionary *device) {
                            [device removeObjectForKey:SentryDeviceContextFreeMemoryKey];
                        }];
}

- (void)modifyDeviceContext:(SentryEvent *)event block:(void (^)(NSMutableDictionary *))block
{
    if (nil == event.context || event.context.count == 0 || nil == event.context[@"device"]) {
        return;
    }

    NSMutableDictionary *context = [[NSMutableDictionary alloc] initWithDictionary:event.context];
    NSMutableDictionary *device =
        [[NSMutableDictionary alloc] initWithDictionary:context[@"device"]];
    block(device);
    context[@"device"] = device;
    event.context = context;
}
```

Reply:
> In the case of an OOM, we'll still get the up-to-date value of free memory with this fix in place

Read about OOMs in detail here https://docs.sentry.io/platforms/apple/configuration/out-of-memory/. OOMs happen on a previous run and we don't know what the free memory was when the watchdog killed the application.

Reply:
> > In the case of an OOM, we'll still get the up-to-date value of free memory with this fix in place
> 
> Read about OOMs in detail here https://docs.sentry.io/platforms/apple/configuration/out-of-memory/. OOMs happen on a previous run and we don't know what the free memory was when the watchdog killed the application.

Right, that makes sense 👍